packages/calcite-components/src/components/rating/rating.tsx

```diff
@@ -407,35 +353,77 @@ export class Rating
       } else if (this.required && numberKey > 0 && numberKey <= this.max) {
         this.value = numberKey;
       }
+      this.updatefocus();
     }
   };
 
   private handleInputChange = (event: InputEvent) => {
     if (this.isKeyboardInteraction === true) {
       const inputVal = Number(event.target["value"]);
-      this.focusValue = inputVal;
       this.hoverValue = inputVal;
       this.value = inputVal;
     }
   };
 
   private handleLabelPointerOver = (event: PointerEvent) => {
-    const target = event.currentTarget as HTMLLabelElement;
-    const newPointerValue = Number(target.firstChild["value"] || 0);
-    this.hoverValue = newPointerValue;
-    this.focusValue = null;
+    this.hoverValue = this.getValueFromLabelEvent(event);
   };
 
   private handleLabelPointerDown = (event: PointerEvent) => {
     const target = event.currentTarget as HTMLLabelElement;
-    const inputVal = Number(target.firstChild["value"] || 0);
+    const inputValue = Number(target.firstChild["value"]);
```

Comment:
maybe instead of using `target.firstChild` you can use the `data-value` attribute?